### DISCUSSION TITLE
Unable to activate Cart Transform Function

### DISCUSSION THREAD

**Discussion by BrandoCommando** [Shopify/function-examples#256] [Feedback: Cart Transform API]
Hello there! Is there something special that needs to be done to activate Cart Transformation functions? I've tried the stock/template Bundling function based on Rust, and I've tried my own version (with JS compiled into Wasm), but I cannot see any evidence that the function is even being run!

I brought this question up on the recent Checkout Extension Q&A sessions, but the only answer I got was that "Bundling is not available to the general public yet"! I am quite aware of the fact that a Partner account is necessary, and development requires a partner development store. I have met those requirements.

I ask if there's something special that must be done because Cart/Checkout validation functions need to be whitelisted, but I haven't found anywhere in the store admin section to whitelist/activate my cart transformation functions. Then, when I check the extension "runs", it is always empty, no matter what I do in the store checkout. All the other function types I have been able to get to work, but the bundling function still eludes me.

Please advise, and thanks for your time.

**Comment by dekstech** [comment]
I am having the same problem as well. The sample app works just fine but when I try to rebuild the function on my own, it isn't working and giving me a ton of errors when I attempt to deploy said extension. Did you try to do npm run deploy to make this extension live with your app?

**Comment by BrandoCommando** [comment]
I've figured it out! You need to run the following mutation to register the extension:
```graphql
mutation cartTransformCreate($functionId: String!) {
  cartTransformCreate(functionId: $functionId) {
    cartTransform {
      functionId
      id
    }
    userErrors {
      field
      message
    }
  }
}
```

**Comment by dekstech** [comment]
@BrandoCommando  can you share your wasm? I can’t get my app to even compile after doing the standard extensions from Shopify…

**Reply by dekstech** [reply]
My npm run deploy throws tons of errors around rust and how something about how everything must be true

**Reply by dekstech** [reply]
https://github.com/Shopify/function-examples/issues/255

**Comment by ignacio-chiazzo** [comment]
Hi! For others reading this issue, please look at the [customized Bundles guide](https://shopify.dev/docs/apps/selling-strategies/bundles/add-a-customized-bundle). We added a section that describes how to register the function in a shop.

**Reply by BrandoCommando** [reply]
Thanks for getting that updated! Seems like something that should be configurable though, like cart validators.
